squidDrone/src/com/com_message_buffer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SRC_COM_MESSAGE_BUFFER_HPP_
+#define SRC_COM_MESSAGE_BUFFER_HPP_
+
+#include <array>
+#include <queue>
+#include "com_types.hpp"
+
+namespace com {
+  /** 
+    * @brief This class represents a queue style buffer for communication
+    * frames exchanged between drones.
+    * 
+    */
+  class ComMessageBuffer {
+    public:
+      /** 
+       * @brief Construct a new Com Message Buffer object
+       * 
+       */
+      ComMessageBuffer(){};
+
+      /** 
+       * @brief Transfer data to queue buffer. 
+       * 
+       * @param data Reference to data frame. Must conform to data frame specifications.
+       * @return types::ComError COM_OK if ok, COM_BUFFER_IO_ERROR if not ok.
+       * 
+       */
+      auto PutData(std::array<std::uint8_t, 32> &data) noexcept -> types::ComError;
+
+      /** 
+       * @brief Retrieve data frame from queue buffer.
+       * 
+       * @return std::array<std::uint8_t, 32> 
+       * 
+       */
+      auto GetData() noexcept -> std::array<std::uint8_t, 32>;
+
+      /**
+       * @brief Check if the buffer is empty.
+       * 
+       * @return true If buffer is empty returns true.
+       * @return false If buffer is not empty returns false.
+       */
+      auto BufferIsEmpty() const noexcept -> bool;
```

Comment:
Is there a use case that this is ever used outside of unittests? To me it sounds like this should be private and is only public because of unittests

Reply:
Yes. Please consider a consumer of the message buffer. How is the consumer to know, if the buffer still contains data? Either have to return a predefined value which can be recoginzed as invalid or give the consumer some means to check the buffer fill status. I decided to use the latter, as it is more efficient to query the buffer than to check if i.e. the array is only filled with '0'.

Reply:
accepted